src/server.c

```diff
@@ -4316,6 +4325,26 @@ int processCommand(client *c) {
     return C_OK;
 }
 
+/* Check if the multiple keys of command are cross slot,
+ * if it is, increment the server stat to record. */
+void checkCommandKeysCrossSlot(client *c) {
+    getKeysResult result = GETKEYS_RESULT_INIT;
+    int numkeys = getKeysFromCommand(c->cmd, c->argv, c->argc, &result);
+    if (numkeys > 1) {
+        int slot = -1;
+        for (int j = 0; j < numkeys; j++) {
+            robj *thiskey = c->argv[result.keys[j].pos];
+            int thisslot = keyHashSlot(thiskey->ptr, sdslen(thiskey->ptr));
+            if (slot == -1) slot = thisslot;
+            if (thisslot != slot) {
+                server.stat_cluster_incompatible_ops++;
```

Comment:
i realize that in scripts running in cluster mode, we also check the compatibility between the declared keys, and the accessed ones (by using the client to remember the slot). i.e we remember the slot from processCommand and match  to it when a command in the script is executed.

and also the compatibility between keys of different commands within the same script (like for MULTI)

i suppose we need to find a way to handle that here too.

Reply:
the accessed keys may are the same with the declared ones, we should handle this case?

Reply:
I think so.. in cluster mode, we do allow acc3ssing undeclared keys, but we restrict them from mismatching the declared ones (if there were any)

Reply:
Hi @oranagra I added a new variable to trace the slot, instead of reuse `client->slot` to avoid confusion in future.